Clockwise/CircuitBreaker.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+using System.Threading.Tasks;
+
+namespace Clockwise
+{
+    public abstract class CircuitBreaker<T> : IObserver<CircuitBreakerStateDescriptor>
```

Comment:
`IObserver<T>` seems a bit strange. Given the way it's implemented, I have to cast it to call `OnNext`, and `OnNext` is a strange method name given the responsibilities of the class. A public method like `SetState` would be clearer.

Reply:
why cast it?